### PR TITLE
asynchronous file handling

### PR DESCRIPTION
**Pull request by KJ002**
Handling io to the file asynchronously as to avoid blocking.

Closing #171

### CODE REVIEW

File: bot/cogs/todo_list.py [hunk 1]
```diff
@@ -176,13 +177,13 @@ async def show_todos(self, ctx: Context) -> None:
             log.info("todo_list_data.json exists")
 
         else:
-            with open(TODO_FILE, "w") as f:
-                f.write(f"{''}")
+            async with aiofiles.open(TODO_FILE, "w") as f:
+                await f.write(f"{''}")
```
Comment (GDWR) [inline bot/cogs/todo_list.py:181]: why have you done `(f"{''}")` over `("")` ?

Comment (ForbiddenKJ) [reply bot/cogs/todo_list.py:181]: Wasn't really looking at the code I was changing, just made sure it was asynchronous and that it worked. Although I will change it.

File: bot/cogs/todo_list.py [hunk 2]
```diff
@@ -222,8 +223,8 @@ async def nuke_todos(self, ctx: Context) -> None:
         if TODO_FILE.is_file():
             if os.stat(TODO_FILE).st_size > 0:
 
-                with open(TODO_FILE, "r+") as read:
-                    todo_file_read = json.load(read)
+                async with aiofiles.open(TODO_FILE, "r+") as read:
+                    todo_file_read = json.load(await read.read())
```
Comment (fisher60) [inline bot/cogs/todo_list.py:227]: this needs to be `json.loads()`, not `json.load()`, I changed it on my end for the merge.

**Review by fisher60 — APPROVED** [review]
This looks great, sorry it took so long to review. There was a minor issue with the todo commands, but I fixed it on my end and left a comment. Well done.

**Pull request merged**